src/main/java/org/cryptomator/ui/addvaultwizard/CreateNewVaultLocationController.java

```diff
@@ -137,12 +133,50 @@ private boolean isActuallyWritable(Path p) {
 
 	@FXML
 	public void initialize() {
-		radioButtonVBox.getChildren().addAll(1, locationPresetBtns); //first item is the list header
-		locationPresetsToggler.getToggles().addAll(locationPresetBtns);
+		var task = backgroundExecutor.submit(this::loadLocationPresets);
+		window.addEventHandler(WindowEvent.WINDOW_HIDING, _ -> task.cancel(true));
 		locationPresetsToggler.selectedToggleProperty().addListener(this::togglePredefinedLocation);
 		usePresetPath.bind(locationPresetsToggler.selectedToggleProperty().isNotEqualTo(customRadioButton));
 	}
 
+	private void loadLocationPresets() {
+		Platform.runLater(() -> loadingPresetLocations.set(true));
+		LocationPresetsProvider.loadAll(LocationPresetsProvider.class) //
+				.flatMap(LocationPresetsProvider::getLocations) //we do not use sorted(), because it evaluates the stream elements, blocking until all elements are gathered
+				.forEach(this::createRadioButtonFor);
+		Platform.runLater(() -> loadingPresetLocations.set(false));
+	}
+
+	private void createRadioButtonFor(LocationPreset preset) {
+		assert !radioButtonVBox.getChildren().isEmpty();
+
+		Platform.runLater(() -> {
+			var btn = new RadioButton(preset.name());
+			btn.setUserData(preset.path());
+
+			//in place sorting
+			var vboxElements = radioButtonVBox.getChildren();
+			boolean added = false;
+			for (int listIndex = 0; listIndex < vboxElements.size(); listIndex++) {
+				if (vboxElements.get(listIndex) instanceof RadioButton rb) {
+					//another radio button
+					if (rb.getText().compareTo(preset.name()) > 0) {
+						vboxElements.add(listIndex, btn);
+						added = true;
+						break;
+					}
+				}
+			}
```

Comment:
I believe there is a simpler approach of adding the button to the right place, relying on existing JavaFX APIs:

```java
// set up once (in `initialize()`), then just add new items to `radioButtons`:
ObservableList<RadioButton> radioButtons = FXCollections.emptyObservableList();
SortedList<RadioButton> sortedButtons = radioButtons.sorted(Comparator.comparing(RadioButton::getText));
Bindings.bindContent(radioButtonVBox.getChildren(), sortedButtons);
```

Reply:
Or use some custom comparator via `this::someMethod`, which puts the custom location button last.

Reply:
Fixed in https://github.com/cryptomator/cryptomator/pull/3236/commits/248d655afb224889769a2c2544954b00be136710